internal/transport/transport_test.go

```diff
@@ -476,10 +486,42 @@ func setUpWithNoPingServer(t *testing.T, copts ConnectOptions, connCh chan net.C
 		}
 		framer := http2.NewFramer(conn, conn)
 		if err := framer.WriteSettings(); err != nil {
-			t.Errorf("Error at server-side while writing settings: %v", err)
+			t.Errorf("Error at server-side while writing settings: %v", framer.ErrorDetail())
 			close(connCh)
 			return
 		}
+		if respModeCh != nil {
+			if err := framer.WriteSettingsAck(); err != nil {
+				t.Errorf("Error while writing settings: %v", err)
+				return
+			}
+			go func() {
+				respondToPing := true
+				var lock sync.Mutex
+				for {
+					frame, err := framer.ReadFrame()
+					if err != nil {
+						return
+					}
+					select {
+					case data, ok := <-respModeCh:
+						if !ok {
+							return
+						}
+						respondToPing = data
```

Comment:
minor nit
```suggestion
					case mdoe, ok := <-respModeCh:
						if !ok {
							return
						}
						respondToPing = mode
```

Reply:
Thanks @arvindbr8 . I didn't use your suggestion directly as there was a typo, but I just made the change you suggested (replace `data` by `mode`).

Reply:
thanks!